hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -446,6 +448,14 @@ Tuple2<CompletableFuture<Void>, String> requestTermination(
             boolean userInitiated
     ) {
         mc.coordinationService().assertOnCoordinatorThread();
+
+        // Job is not allowed to be suspended of it is not suspendable in config
+        // and termination mode is SUSPEND or RESTART.
+        if (!Util.isJobSuspendable(mc.jobConfig()) && mode.actionAfterTerminate() != CANCEL) {
```

Comment:
`isJobSuspendible()` is only used in `MasterJobContext`, so it should be moved to that class. It can even be removed completely by the introduction of `JobConfig.getArgumentOrDefault()`.
```suggestion
        ActionAfterTerminate action = mode.actionAfterTerminate();
        if ((action == SUSPEND || action == RESTART) && !isJobSuspendable(mc.jobConfig())) {
```

Reply:
No, it is not. Take a look at https://github.com/hazelcast/hazelcast-enterprise/pull/6812